lib/tree.rb

```diff
@@ -1,38 +1,61 @@
 class NoApplesError < StandardError; end
 
+
 class AppleTree
-  attr_#fill_in :height, :age, :apples, :alive
+  attr_accessor :height, :age, :apples, :alive
 
   def initialize
+    @height = 0
+    @age = 0
+    @apples = []
+    @alive = true
+    @colors = ['red', 'green', 'yellow']
   end
 
   def age!
+    @age += 1
+    @height += rand(1..(2*@age))
+    self.add_apples
   end
 
   def add_apples
+    rand(1..(4*@age)).times do
+      @apples << Apple.new(@colors[rand(0..2)], rand(1..5))
+    end
   end
 
   def any_apples?
+    @apples.length > 0 ? true : false
   end
 
   def pick_an_apple!
     raise NoApplesError, "This tree has no apples" unless self.any_apples?
+    @apples.slice!(rand(0...@apples.length))
   end
 
   def dead?
+    if @height > 60 || @age > 13
+      true
+    else
+      false
+    end
```

Comment:
This could just be `@height > 60 || @age > 13`, right?
But what I'm really wondering is why it isn't just `!@alive`

Reply:
I like the simplicity of `@height > 60 || @age > 13`. Since `@alive` isn't used or necessary I think I'll just remove it